core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt

```diff
@@ -125,45 +119,58 @@ fun LpEditFloatingActionButton(
 }
 
 /**
- * Create [BadgeButton] compose for user's notifications
+ * Function that creates [LpBadgeButton] compose for user's notifications
  *
  * @param modifier to modify box properties that contains: Button & Badge
  * @param badgeNumber to modify the number of notifications shown
+ * @param showBadge to enable the number of notifications to be displayed
+ * @param imageVector to modify the icon to be displayed
  * @param onClick high order function, to define button action
  *
  * @author Jose Miguel Garcia Reyes
  */
+
 @ExperimentalMaterial3Api
 @Composable
-fun BadgeButton(
+fun LpBadgeButton(
     modifier: Modifier = Modifier,
-    onClick: () -> Unit,
-    badgeNumber: Int
+    badgeNumber: Int = 0,
+    showBadge: Boolean = false,
+    imageVector: ImageVector = Icons.Filled.Notifications,
+    onClick: () -> Unit = {}
 ) {
-    Box {
+    Box(modifier = modifier) {
         FilledIconButton(
             onClick = onClick,
-            colors = IconButtonDefaults.filledIconButtonColors(MintJulep)
+            colors = IconButtonDefaults.filledIconButtonColors(MaterialTheme.colorScheme.primary)
         ) {
             Icon(
-                Icons.Filled.Notifications,
-                contentDescription = "Notifications button",
-                tint = Drab
+                imageVector = imageVector,
+                contentDescription = "icon image",
+                tint = MaterialTheme.colorScheme.onPrimary
             )
         }
-
-        if (badgeNumber > 0) {
-            BadgedBox(modifier = modifier
-                .padding(horizontal = 15.dp, vertical = 7.dp)
-                .align(Alignment.BottomEnd), badge = {
-                Badge(
-                    modifier = Modifier
-                        .padding(all = 1.dp)
-                        .align(Alignment.TopEnd)
-                ) {
-                    Text(text = badgeNumber.toString())
+        if (showBadge) {
+            BadgedBox(
+                modifier = Modifier
+                    .padding(
+                        horizontal = 15.dp,
+                        vertical = 7.dp
+                    )
+                    .align(alignment = Alignment.BottomEnd),
+                badge = {
+                    Badge(
+                        modifier = Modifier
+                            .padding(all = 1.dp)
+                            .align(alignment = Alignment.TopEnd)
+                    ) {
+                        Text(
+                            text = badgeNumber.toString(),
```

Comment:
What do you think if we change badgeNumber to String and we avoid toString cause it is unnecessary if it will always be cast

Reply:
Agree, already changed